src/tracker.c

```diff
@@ -1190,18 +1190,28 @@ void editor_command(int id) {
 	case ID_TRANSPOSE: {
 		int delta = DialogBox(hinstance, MAKEINTRESOURCE(IDD_TRANSPOSE),
 			hwndMain, TransposeDlgProc);
-		if (delta == 0) break;
-		for (BYTE *p = sel_start; p < sel_end; p = next_code(p)) {
-			int note = *p - 0x80;
-			if (note < 0 || note >= 0x48) continue;
-			note += delta;
-			note %= 0x48;
-			if (note < 0) note += 0x48;
-			*p = 0x80 + note;
+		if (delta != 0) {
+			for (BYTE *p = sel_start; p < sel_end; p = next_code(p)) {
+				BYTE *note_ptr = NULL;
+				if (*p == 0xF9) {
+					// F9 is a pitch bend to a specific note
+					note_ptr = &p[3];
+				} else if (*p >= 0x80 && *p < 0xC8) {
+					// This is a command to play a given note
+					note_ptr = p;
+				}
+				if (note_ptr) {
+					int note = *note_ptr - 0x80;
+					note += delta;
+					note %= 0x48;
+					if (note < 0) note += 0x48;
+					*note_ptr = 0x80 + note;
+				}
```

Comment:
Since `0xF9` operates on an absolute note, I think the `if (note_ptr) {` logic should be moved into the `else if` and special handling added for `0xF9` that doesn't wrap to the normal 0x80-0xC8 range since the note could fall outside of that range. I think wrapping to 0x00-0xFF should be fine since it doesn't make any assumptions as to what the user is trying to do other than a basic transposition.

Reply:
Should be adjusted to do that now